arch/arm/core/aarch32/mpu/arm_mpu_v8_internal.h

```diff
@@ -28,28 +28,124 @@ struct dynamic_region_info {
  * regions may be configured.
  */
 static struct dynamic_region_info dyn_reg_info[MPU_DYNAMIC_REGION_AREAS_NUM];
+#if defined(CONFIG_CPU_CORTEX_M23) || defined(CONFIG_CPU_CORTEX_M33) || \
+	defined(CONFIG_CPU_CORTEX_M55)
+static inline void mpu_set_mair0(uint32_t mair0)
+{
+	MPU->MAIR0 = mair0;
+}
+
+static inline void mpu_set_rnr(uint32_t rnr)
+{
+	MPU->RNR = rnr;
+}
```

Comment:
```
__ASSERT(function_names_should_provide_meaningful_abstraction())
maybe_just_use_the_struct_directly()
save_30_lines_too();
```

Reply:
> maybe_just_use_the_struct_directly()
save_30_lines_too();

The idea of this MPU internal abstraction is to isolate the specifics for Armv8-M and ARMv8-R AArch32 in order to reuse the rest of the common code. For ARMv8-R AArch32, the MPU registers must be accessed using the `mrc/mcr` hence is not possible to use the same implementation as for Armv8-M.